enigma/src/main/java/cuchaz/enigma/utils/validation/PrintValidatable.java

```diff
@@ -1,34 +1,47 @@
 package cuchaz.enigma.utils.validation;
 
+import org.tinylog.Level;
+import org.tinylog.Logger;
+
 import java.io.PrintStream;
 import java.util.Arrays;
 
 public class PrintValidatable implements Validatable {
-
 	public static final PrintValidatable INSTANCE = new PrintValidatable();
 
 	@Override
 	public void addMessage(ParameterizedMessage message) {
-		formatMessage(System.out, message);
+		formatMessage(Level.INFO, message);
 	}
 
-	public static void formatMessage(PrintStream w, ParameterizedMessage message) {
+	public static void formatMessage(Level level, ParameterizedMessage message) {
 		String text = message.getText();
 		String longText = message.getLongText();
 		String type = switch (message.message.type) {
 			case INFO -> "info";
 			case WARNING -> "warning";
 			case ERROR -> "error";
 		};
-		w.printf("%s: %s\n", type, text);
+
+		log(level, "{}: {}\n", type, text);
 
 		if (!longText.isEmpty()) {
-			Arrays.stream(longText.split("\n")).forEach(s -> w.printf("  %s\n", s));
+			Arrays.stream(longText.split("\n")).forEach(s -> log(level, "  {}\n", s));
 		}
 	}
 
 	@Override
 	public void clearMessages() {
+		// no-op
 	}
 
+	private static void log(Level level, String message, Object... args) {
```

Comment:
Does `Logger` not have a `#log(Level, String, Object...)` method?

Reply:
if it does I somehow missed it

Reply:
checked docs, I can confirm that there is none